### PR TITLE
math/Utils: removed, removes support for I/O BigEndian files

### LINKED ISSUE #103: Remove the #define macros in utils/Math.hpp
it's hell ugly. `ITER_1` etc , ideally all of them.

### PR DESCRIPTION
**Pull request by breznak**
- we do not support any BigEndian platforms, that code has not been run
or tested
- otherwise, math/Utils had no usage, so can be removed

Fixes #103 
For #306 VectorFile: removes BigEndian support 

API breakage: VectorFileSensor can no longer Read files created on BigEndian on a LittleEnd platform, but since we revamped Serialization completely by removing capnp, this is not an issue. As now we support serialization only on the same platform as the code runs.

### CODE REVIEW

**Comment by breznak** [comment]
@dkeeney you should probably review this, as you made the capnp switch and work on NetworkAPI.

File: src/nupic/regions/VectorFile.cpp [hunk 1]
```diff
@@ -29,7 +29,6 @@
 #include <cstdio> //fopen
 #include <iostream>
 #include <math.h>
-#include <nupic/math/Utils.hpp> // For isSystemLittleEndian and utils::swapBytesInPlace.
```
Comment (breznak) [inline src/nupic/regions/VectorFile.cpp]: these were the only uses of math/Utils,
and I could remove the support (for reading files with different endian-ness, as we can only serialize to/from files on the same platform now)

**Comment by ctrl-z-9000-times** [comment]
See also issue #175.

**Comment by dkeeney** [comment]
>  I don't fully understand the VectorFile code so I can't approve.

The code is actually rather ugly and there is no reason that it needs to be that way.  But it works and I think it is a lower priority to clean up that code.  Perhaps a cleanup would be appropriate if/when we change the name as per  Issue #306.

**Comment by breznak** [comment]
Follow up would be #306 and mainly #135

**Pull request merged**